src/ride/ride.c

```diff
@@ -183,9 +183,17 @@ rct_ride_measurement *get_ride_measurement(int index)
 	return &(RCT2_ADDRESS(RCT2_ADDRESS_RIDE_MEASUREMENTS, rct_ride_measurement)[index]);
 }
 
-rct_ride_type *ride_get_entry(rct_ride *ride)
+rct_ride_type *get_ride_entry_by_ride(rct_ride *ride)
 {
-	return get_ride_entry(ride->subtype);
+	rct_ride_type *type = get_ride_entry_by_index(ride->subtype);
+	if (type == NULL)
+	{
+		char oldname[128];
+		format_string(oldname, ride->name, &ride->name_arguments);
+		oldname[sizeof(oldname) - 1] = '\0';
```

Comment:
I assume this is to stop it from overrunning during log_error if the ride name is longer than 127 characters but by this point damage will already have been done since format_string will have written over the memory boundary.